code/android-core-library/src/main/java/com/adobe/marketing/mobile/internal/eventhub/history/AndroidEventHistoryDatabase.java

```diff
@@ -47,19 +49,14 @@ class AndroidEventHistoryDatabase implements EventHistoryDatabase {
      */
     AndroidEventHistoryDatabase() throws EventHistoryDatabaseCreationException {
         try {
-            //TODO: we will create a utility method: Context.getDatabasePath(), we need to  refactor the following code after that.
-            final File applicationCacheDir = ServiceProvider.getInstance().getDeviceInfoService().getApplicationCacheDir();
-            if (applicationCacheDir != null) {
-                final String cacheDirCanonicalPath = applicationCacheDir.getCanonicalPath();
-                databaseFile = new File(cacheDirCanonicalPath + "/" + DATABASE_NAME);
-            }
+            openOrMigrateEventHistoryDatabaseFile();
```

Comment:
Remove all this try/catch blocks. 

databaseFile = openOrMigrateEventHistoryDatabaseFile(); 
if (dataFile == null) {
 // throw EventHistoryDatabaseCreationException 
}

if (!SQLiteDatabaseHelper.createTableIfNotExist(databaseFile.getCanonicalPath(), tableCreationQuery)) {
   // throw  EventHistoryDatabaseCreationException
}

Reply:
`getCanonicalPath` throws `IOException`, need a try/catch block for that